### PR TITLE
feat(validate/pubrules): add pubrules validator

### PR DESCRIPTION
**Pull request by sidvishnoi**
[WIP]
Closes https://github.com/w3c/spec-prod/issues/3

### CODE REVIEW

**Comment by marcoscaceres** [comment]
Do you remember what else we needed to do here?

**Comment by sidvishnoi** [comment]
1. Link it properly with build output.
2. Figure out `W3C_API_KEY` handling.
3. Handle errors properly.
4. Cleanup dev/debug stuff.

**Comment by sidvishnoi** [comment]
I just saw we've TODO comments in code for exactly that 😄

**Review by marcoscaceres — APPROVED** [review]

File: test/validate-pubrules.test.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+export default async function validatePubrules(outputs: Outputs) {
+	const { pubrules: shouldValidate = false } = outputs?.prepare?.validate || {};
+	if (shouldValidate === false) {
+		return;
+	}
+
+	const { dest = process.cwd() + ".w3c", file = "index.html" } =
+		outputs?.build?.w3c || {};
+
+	return await main({ dest, file });
```
Comment (sidvishnoi) [inline test/validate-pubrules.test.ts:13]: Needs W3C_API_KEY as input, but let's ignore it for now.

Comment (sidvishnoi) [reply test/validate-pubrules.test.ts]: Done.

**Pull request merged**